runtime/caml/mlvalues.h

```diff
@@ -88,6 +88,30 @@ typedef uintnat mark_t;
 #define Is_exception_result(v) (((v) & 3) == 2)
 #define Extract_exception(v) ((v) & ~3)
 
+/* Pointers outside the OCaml heap that are used as values.
+   (E.g. code pointers.)  Must be 2-aligned. */
+
+Caml_inline value Val_foreign_ptr(void * p)
```

Comment:
I thought it would be useful for performance comparison purposes to preserve the current implementation (using unmodified naked pointers) in "naked pointers" mode.  Eventually, we would keep only the "+1 / -1" implementation.